api/src/main/java/com/epam/pipeline/manager/search/SearchRequestBuilder.java

```diff
@@ -183,4 +184,17 @@ private void addAclFilters(final BoolQueryBuilder queryBuilder, final PipelineUs
                         .orElseThrow(() -> new SearchException("Missing index name for type: " + type)))
                 .toArray(String[]::new);
     }
+
+    private String escapeUnsupportedSpecialCharacters(final String string) {
+        final StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < string.length(); i++) {
+            final char c = string.charAt(i);
+            if (c == '\\' || c == '!' || c == ':' || c == '^' || c == '[' || c == ']' || c == '{' || c == '}'
```

Comment:
Could you please add all special characters to a set? E.g. `Set<Character> special = new HashSet<>(Arrays.asList('\\', '!'))`; if (special.contains(c)) ......;

Reply:
Done